ssz-rs/src/list.rs

```diff
@@ -150,6 +150,7 @@ where
     type Output = <Idx as SliceIndex<[T]>>::Output;
 
     fn index(&self, index: Idx) -> &Self::Output {
+        // Index::index implementations may panic when the given index is out of bounds
```

Comment:
I would lean towards leaving these comments out as the idea is to also satisfy the `Index::index` interface here but if you want them in, we can keep them

ditto for the rest of these that just lean on the wrapping interface for invariants